src/workloads/execution/ShardedLookup.yml

```diff
@@ -0,0 +1,116 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+
+GlobalDefaults:
+  Nop: &Nop {Nop: true}
+
+  Database: &Database test
+
+  LocalCollection: &LocalCollection Collection0
+  LocalNamespace: &LocalNamespace test.Collection0
+
+  ForeignCollection: &ForeignCollection Collection1
+  ForeignNamespace: &ForeignNamespace test.Collection1
+
+  DocumentCount: &DocumentCount 5000
+  BatchSize: &BatchSize 1000
+  NumChunks: &NumChunks 5
+
+Actors:
+- Name: CreateShardedCollections
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationMetricsName: EnableSharding
+      OperationName: AdminCommand
+      OperationCommand:
+        enableSharding: *Database
+    - OperationMetricsName: ShardLocalCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *LocalNamespace
+        key: {shardKey: hashed}
+        numInitialChunks: *NumChunks
+    - OperationMetricsName: ShardForeignCollection
+      OperationName: AdminCommand
+      OperationCommand:
+        shardCollection: *ForeignNamespace
+        key: {shardKey: hashed}
+        numInitialChunks: *NumChunks
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: LoadInitialData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    BatchSize: *BatchSize
+    Threads: 1
+    DocumentCount: *DocumentCount
+    Database: *Database
+    CollectionCount: 2    # Loader will populate 'Collection0' then 'Collection1'.
+    Document:
+      shardKey: {^RandomInt: {min: 1, max: 100}}
+      date: &Date {^RandomDate: {min: "2020-01-01", max: "2021-01-01"}}
+      int: {^RandomInt: {min: 1, max: 100}}
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: admin
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        fsync: 1
+  - *Nop
+
+- Name: RunLookups
+  Type: RunCommand
+  Database: *Database
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 10    # Untargeted $lookup from sharded collection to sharded collection
+    Database: *Database
+    Operations:
+    - OperationMetricsName: UntargetedLookupShardedToSharded
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *LocalCollection
+        pipeline:
+          [{
+            $lookup: {
+              from: *ForeignCollection,
+              let: {localInt: "$int"},
+              pipeline: [{
+                $match: {
+                  $expr: {
+                    $and: [
+                      {$eq: ["$int", "$$localInt"]},
+                      {$lte: ["$date", *Date]}
```

Comment:
Is this `Date` always going to be the last date that is set on line 60? If so, I can see there being a lot of variability in this workload depending on what the last `Date` was. Is there a reason we aren't using something like `$$localDate` instead, similar to `$$localInt`?

Reply:
No, `*Date` just represents the yaml value of `Date`, so it will generate a new random date for each $lookup we run. We can see this in the logs, which show the pipelines that are generated; each pipeline has a different date value. 

I just wanted to avoid running identical queries on the foreign collection. If we went with something like `$$localDate`, then the subpipelines (and the top-level $lookup) would be identical on each run.

Reply:
Ah that makes sense about `*Date` generating a new date every time - thanks for the clarification! I don't think it's necessarily a bad thing for the repeated runs to generate the same $lookup subpipeline for each document if we use `$$localDate`, since we'd then have smaller variance in the repeat performance and thus a stronger indication if there were ever a code change that caused perf regressions (although maybe more variance between runs since the performance for each run is more contingent on the generated documents). In saying that, I'm also fine with the `*Date` method you're using, and the variance between runs is looking good; feel free to keep this as is.

Reply:
Thanks for the explanation! I tried running with `$$localDate` instead and got nearly identical results. I'm going to keep it as-is for now, but may reconsider as we add new phases to this workload.